### PR TITLE
Feat: Add tests and functionality for user story 8

### PR DESCRIPTION
**Pull request by MatthewTLim**
# Describe the changes below:
 This PR will add all functionality for User story 8
# Relevant user story:
User Story 8
# Specific feedback requests? Include file name and line:
please look at the way I am implementing the flash message. Just curious if there is another way to approach this. Flash message is in the file views/items/show.html.erb
# Before submitting, check the following:
- [x] Entire test suite is passing
- [99.54%] SimpleCov test percentage

# Note any co-authors / anything else you'd like to add:

### CODE REVIEW

File: app/views/items/show.html.erb [hunk 1]
```diff
@@ -1,3 +1,8 @@
...
-<%= @item.description %>
-<%= @item.unit_price %>
\ No newline at end of file
+<p>Description: <%= @item.description %></p>
+<p>Unit Price: <%= @item.unit_price %></p>
+<%= link_to "Edit #{@item.name}", edit_merchant_item_path(@merchant, @item) %>
+<div class="flash-message">
+<% if flash[:notice] %>
+  <p><%= flash[:notice] %></p>
+<% end %>
```
Comment (westonio) [inline app/views/items/show.html.erb:8]: Looks good!

Just a note for later refactoring - we can have an each statement in the app/views/layouts/application file that will take care of displaying flash messages on all pages.

**Review by westonio — APPROVED** [review]

File: app/views/items/show.html.erb (hunk 1 above)
Comment (MatthewTLim) [reply app/views/items/show.html.erb:8]: Awesome this is exactly what I was wondering. Thank you for the feedback!

File: app/controllers/items_controller.rb [hunk 2]
```diff
@@ -4,7 +4,31 @@ def index
...
+  def edit
+    @merchant = Merchant.find(params[:merchant_id])
+    @item = Item.find(params[:id])
+  end
+
+  def update
+    @merchant = Merchant.find(params[:merchant_id])
+    @item = Item.find(params[:id])
+    if @item.update(item_params)
+      flash[:notice] = "#{@item.name} has been updated!"
```
Comment (dani-wilson) [inline app/controllers/items_controller.rb:20]: I like the interpolation in this flash notice!

File: config/routes.rb [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Rails.application.routes.draw do
   resources :merchants, only: [] do
     resources :dashboard, only: :index
-    resources :items, only: [:index, :show]
+    resources :items, only: [:index, :show, :edit, :update]
```
Comment (dani-wilson) [inline config/routes.rb:6]: 😄

**Review by dani-wilson — APPROVED** [review]
looks great!

**Pull request merged**